src/manager/agent.manager.service.ts

```diff
@@ -33,7 +34,7 @@ export class AgentManagerService {
      * TODO need to think through a few more cases - like how the public endpoints and ports will work
      * TODO need to handle error cases and ensure logging works in our deployed envs
      */
-    public async spinUpAgent(walletId: string, walletKey: string, adminApiKey: string, ttl?: number, seed?: string, controllerUrl?: string, alias?: string) {
+    public async spinUpAgent(walletId: string, walletKey: string, adminApiKey: string, ttl?: number, seed?: string, controllerUrl?: string, alias?: string, autoConnect: boolean = true) {
         const runningData = await this.handleAlreadyRunningContainer(alias, adminApiKey);
```

Comment:
I think you'll need to pass the autoConnect bool into the handleAlreadyRunningContainer as that also makes a call to create the connection